src/builder/modules/Generator.res

```diff
@@ -143,6 +147,72 @@ let drawLine = (
     Generator_Builder.drawLine(model.contents, from, to, ~color, ~width, ~pattern=[], ~offset=0)
 }
 
+let drawLinePath = (
+  points: array<Builder.position>,
+  ~color: string="#000000",
+  ~width: int=1,
+  /* ~pattern: array<int>=[],
+   ~offset: int=0, */
+  ~close: bool=false,
+  (),
+) => {
+  model :=
+    Generator_Builder.drawPath(
+      model.contents,
+      points,
+      ~color,
+      ~width,
+      ~pattern=[],
+      ~offset=0,
+      ~close,
+    )
+}
+
+let drawLineRect = (
+  dest: Builder.rectangle,
+  ~color: string="#000000",
+  ~width: int=1,
+  /* ~pattern: array<int>=[],
+   ~offset: int=0, */
+  (),
+) => {
+  let (x, y, w, h) = dest
+  model :=
+    Generator_Builder.drawPath(
+      model.contents,
+      [(x, y), (x + w, y), (x + w, y + h), (x - 1, y + h), (x - 1, y)],
+      ~color,
+      ~width,
+      ~pattern=[],
+      ~offset=0,
+      ~close=false,
+    )
+}
+
+let drawLineCuboid = (
```

Comment:
Drawing cuboids is a very interesting idea! Is there any particular use case you have in mind for drawing a line cuboid?

Reply:
This and drawLineRect() are mostly for consistency, so that the functions don't only exist for fold lines, but a specific use could be possibly for color coding similar looking parts with outlines, such as for the squid. these could go underneath the fold lines and still be visible through the dashes, and both could be toggleable on or off.

Reply:
> This and drawLineRect() are mostly for consistency, so that the functions don't only exist for fold lines

Yes, I had a feeling that might have been the case.

> but a specific use could be possibly for color coding similar looking parts with outlines, such as for the squid. these could go underneath the fold lines and still be visible through the dashes, and both could be toggleable on or off.

I hadn't thought of that, great creative thinking. I understand that's just an example, but for that case it's probably an improvement we could make to the `drawFoldLine()` function - standardising how coloured fold lines might behave.